msu/hooks/skills/skill.nut

```diff
@@ -310,6 +314,31 @@
 		return ret;
 	}
 
+	local removeSelf = o.removeSelf;
+	o.removeSelf = function()
+	{
+		if (this.isStacking()) return removeSelf();
+
+		if (--this.m.MSU.AddedStack <= 0) return removeSelf();
+
+		if (::MSU.isNull(this.getItem()) || ::MSU.isNull(this.getItem().getContainer()) || this.getItem().getContainer().getActor().getID() != this.getContainer().getActor().getID())
```

Comment:
The `::MSU.isEqual` you talked about was about an item comparison because there can be multiple items on the same actor which have the same id (items in bags for example). Now we are doing an actor comparison and I don't see the need for an `isEqual` comparison as all actors have unique ids. Vanilla does this everywhere in the code for uniquely identifying actors - therefore it is in my opinion as reliable as it should be (otherwise what about all that vanilla code's reliabilty?) and more "familiar" and therefore less complicated.